### DISCUSSION TITLE
How to optimize the plot settings for a workstations?

### DISCUSSION THREAD

**Discussion by pluto2021** [Chia-Network/chia-blockchain#1981] [Q&A]
For workstations that have large memories and typical TB sized hard drives supporting 2 CPUs with multiple cores, how to calculate the optimal settings for the plotting to minimize the time to complete a plot?

**Comment by cccat6** [comment]
From my personal experience.
Plotting time is highly related to the CPU frequency, architecture, and the speed of temporary SSD.
Each plotting task will suggest using 4 threads. It will take half of the CPU time in the process. So every 4 threads can support 2 plotting tasks in optimal.
CPU frequency and architecture really matters. Compare with ZEN1 2.0GHz with ZEN2 4.0GHz, there is more than a 4x time difference between them.
The speed of SSD, in ZEN1 2.0GHz (EPYC7551), each plotting optimal speed needs 80-150MBw/sec. In ZEN2 4.0GHz (3700X), each plotting optimal speed needs 400-600MBw/sec.
The optimal setting for your device really needed to do some experiments. You can try different settings and find the best.
By the way, in all the cases I see. Server CPUs have the worst performance in plotting. Desktop CPUs are usually much better. I guess it based on the speed of memory exchange. If you got a multi-CPU platform, I guess it will get even worse performance because it may cause many memory access latency. You can try anyway.
